package/htmd-deps/write_meta_yaml.py

```diff
@@ -25,18 +25,26 @@
 deps = []
 with open(os.path.join(workdir, 'DEPENDENCIES'), 'r') as f:
     for line in f:
-        deps.append(line.strip().lower())
+        # Remove comments and split
+        packagematch = line.split('#')[0].split()
+        # Append dict of the conda package match specification
+        if packagematch:
+            deps.append(dict(zip(['name', 'version', 'build_string'], packagematch)))
 
 # Get all installed packages from conda
 packages = check_output(['conda', 'list', '--json']).decode('utf8')
 packages = json.loads(packages)
 
 # Find the version of each dependency and add them to the meta.yaml file
-found = {key: False for key in deps}
+found = {dep['name']: False for dep in deps}
 for p in packages:
     name = p['name'].lower()
-    if name in deps:
-        text += '    - {} >={}\n'.format(name, p['version'])
+    if name in [dep['name'] for dep in deps]:
```

Comment:
You could do `if name in list(found.keys())` no? To not redo the list/dict comprehension

Reply:
Or just `if name in found` actually I think works

Reply:
using `if name in found` here is confusing, I think. I've made a new proposal to avoid redoing list comprehension, I hope you approve :)